Add CLUE Benchmark (11 datasets)

JetRunner opened the pull request:
Add 11 tasks of [CLUE](https://github.com/CLUEbenchmark/CLUE).

lhoestq commented on datasets/clue/clue.py:
```diff
@@ -0,0 +1,486 @@
+            name="chid",
+            description=textwrap.dedent(
+                """\
+            Delta Reading Comprehension Dataset (DRCD) belongs to the general field of traditional
+            Chinese machine reading comprehension data set. This data set is expected to become a
+            standard Chinese reading comprehension data set suitable for transfer learning.
```
This is the description of `dcrd`, not `chid` ^^
Maybe a copy/paste issue ?

lhoestq commented on datasets/clue/clue.py:
```diff
@@ -0,0 +1,486 @@
+            name="c3",
+            description=textwrap.dedent(
+                """\
+            Delta Reading Comprehension Dataset (DRCD) belongs to the general field of traditional
+            Chinese machine reading comprehension data set. This data set is expected to become a
+            standard Chinese reading comprehension data set suitable for transfer learning.
```
same here

lhoestq commented on datasets/clue/clue.py:
```diff
@@ -0,0 +1,486 @@
+        test_split = nlp.SplitGenerator(
+            name=nlp.Split.TEST,
+            gen_kwargs={
+                "data_file": os.path.join(data_dir or "", "test.json" if self.config.name != "diagnostics" else "diagnostics_test.json"),
```
Maybe replace `data_dir or ""` by `data_dir`, as `data_dir` is never None ?

lhoestq commented on datasets/clue/clue.py:
```diff
@@ -0,0 +1,486 @@
+                            # For some tasks, the label is represented as 0 and 1 in the tsv
+                            # files and needs to be cast to integer to work with the feature.
+                            if label_classes and label not in label_classes:
+                                print(label, label_classes)
```
Could you remove this `print` statement ?

lhoestq reviewed (commented):
This is awesome ! Thank you so much for adding this one :)

Maybe we can use `ClassLabel` feature type for labels, as in `glue` ? What do you think ?

I also left a few (minor) comments:

JetRunner commented:
Thanks, @lhoestq! I've addressed the comments. 
Also, I have tried to use `ClassLabel` [when possible](https://github.com/huggingface/nlp/pull/572/files#diff-1026ac7d7b78bf029cb0ebe63162c77dR297). Is there still somewhere else we can use `ClassLabel`?

JetRunner commented:
I believe CI failure is unrelated.

brightmart commented:
Great job!

lhoestq reviewed (approved):
Looks good to me, thanks !
Don't pay attention to the CI error, it comes from master and we're currently trying to fix it
